### PR TITLE
[ISSUE-5][ISSUE-6] Added Report module and refactored packages

### PR DESCRIPTION
**Pull request by mssolomonov**
- Added class `MotifReporter` for printing results
- Moved classes for PWM calculation into `ru.spbstu.calculator` package
- Added Main method for demo

### CODE REVIEW

File: src/main/java/ru/spbstu/Main.java [hunk 1]
```diff
@@ -1,11 +1,29 @@
...
+        FastaFile file = new FastaFile();
+        try {
+            file.readFastaFile(FASTA_FILENAME);
+        } catch (IOException e) {
+            System.err.println(e.toString());
+            System.exit(0);
+        }
+        PWCalculator pwCalculator = new PWCalculator();
+        PWMatrix matrix = pwCalculator.calculateMatrix(file, new CalculationStrategy(Constants.frequency,
+                Constants.DEFAULT_WINDOW_SIZE, "loglikelihood"));
```
Comment (darina-ovis) [inline src/main/java/ru/spbstu/Main.java:23]: Use "loglikelihood" as a enum

File: src/main/java/ru/spbstu/Main.java [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
+
+
+public class Main {
+    private static final String FASTA_FILENAME = "src/main/resources/PWMSample";
     public static void main(String[] args) {
-        System.out.println(new Main().getGreeting());
+        FastaFile file = new FastaFile();
+        try {
+            file.readFastaFile(FASTA_FILENAME);
+        } catch (IOException e) {
```
Comment (darina-ovis) [inline src/main/java/ru/spbstu/Main.java:17]: seems redundant

File: src/main/java/ru/spbstu/reporter/MotifReporter.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+import ru.spbstu.calculator.DatasetScore;
+
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.Writer;
+import java.util.Comparator;
+
+public class MotifReporter {
+    public static void reportTopFive(DatasetScore result, PrintStream output) {
+        result.getData().stream().sorted(Comparator.reverseOrder()).limit(5).forEach(output::println);
```
Comment (darina-ovis) [inline src/main/java/ru/spbstu/reporter/MotifReporter.java:12]: get rid of magic numbers

File: src/main/java/ru/spbstu/calculator/CalculationStrategy.java [hunk 4]
```diff
@@ -14,7 +14,7 @@
     public String resultFrequency;
 
 
-    CalculationStrategy(double frequencyOfNucleotides, int sequenceLength, String resultFrequency) {
+    public CalculationStrategy(double frequencyOfNucleotides, int sequenceLength, String resultFrequency) {
```
Comment (darina-ovis) [inline src/main/java/ru/spbstu/calculator/CalculationStrategy.java:17]: add setter/getter or builder

File: src/main/java/ru/spbstu/Main.java (hunk 1 above)
Comment (mssolomonov) [reply src/main/java/ru/spbstu/Main.java:23]: Done

File: src/main/java/ru/spbstu/Main.java (hunk 2 above)
Comment (mssolomonov) [reply src/main/java/ru/spbstu/Main.java:17]: Done

File: src/main/java/ru/spbstu/reporter/MotifReporter.java (hunk 3 above)
Comment (mssolomonov) [reply src/main/java/ru/spbstu/reporter/MotifReporter.java:12]: Done

File: src/main/java/ru/spbstu/calculator/CalculationStrategy.java (hunk 4 above)
Comment (mssolomonov) [reply src/main/java/ru/spbstu/calculator/CalculationStrategy.java:17]: Done

File: src/main/java/ru/spbstu/calculator/CalculationStrategy.java [hunk 5]
```diff
@@ -10,14 +10,52 @@
...
+            return this;
+        }
+
+        public Builder calculationType(calculationMethod value) {
+            this.calculationType = value;
+            return this;
+        }
+
+        public CalculationStrategy build() {
+            return new CalculationStrategy(this.frequencyOfNucleotides, this.sequenceLength, this.calculationType);
```
Comment (darina-ovis) [inline src/main/java/ru/spbstu/calculator/CalculationStrategy.java:38]: minor:
if the building class requires many parameters, it is better to get them from Builder object. For example, `new CalculationStrategy(this);` and the constructor of CalculationStrategy looks like that
```
private CalculationStrategy(Builder builder) {
         this.frequencyOfNucleotides = builder.frequencyOfNucleotides;
         this.sequenceLength = builder.sequenceLength;
         this.resultFrequency = builder.resultFrequency;
     }
```

**Review by darina-ovis — APPROVED** [review]

File: src/main/java/ru/spbstu/calculator/CalculationStrategy.java (hunk 5 above)
Comment (mssolomonov) [reply src/main/java/ru/spbstu/calculator/CalculationStrategy.java:38]: Ok, may change to this later, if more parameters will be required

**Pull request merged**